File: client/src/types/order.ts
export type OrderStatus = 'Pending' | 'Confirmed' | 'Shipped' | 'Delivered' | 'Cancelled'
export type PaymentMethod = 'Cod' | 'MockPaid'

export interface OrderItem {
  productId: string
  productName: string
  imageUrl: string | null
  variantDescription: string | null
  unitPrice: number
  quantity: number
  lineTotal: number
}

export interface OrderStatusHistoryEntry {
  status: OrderStatus
  createdAt: string
}

export interface Order {
  id: string
  status: OrderStatus
  paymentMethod: PaymentMethod
  isPaid: boolean
  shippingAddress: string
  subtotal: number
  discountAmount: number
  voucherCode: string | null
  totalAmount: number
  createdAt: string
  items: OrderItem[]
  statusHistory: OrderStatusHistoryEntry[]
}

export interface OrderSummary {
  id: string
  status: OrderStatus
  totalAmount: number
  itemCount: number
  createdAt: string
  items: OrderItem[]
}

export interface CreateOrderRequest {
  shippingAddress: string
  paymentMethod: PaymentMethod
  voucherCode?: string | null
}

export interface PagedResult<T> {
  items: T[]
  totalCount: number
  page: number
  pageSize: number
  totalPages: number
}

File: src/Domain/Entities/Order.cs
using ShopeeClone.Domain.Common;
using ShopeeClone.Domain.Enums;

namespace ShopeeClone.Domain.Entities;

public class Order : BaseEntity
{
    public string UserId { get; set; } = string.Empty;
    public string ShippingAddress { get; set; } = string.Empty;
    public OrderStatus Status { get; set; } = OrderStatus.Pending;
    public PaymentMethod PaymentMethod { get; set; }
    public bool IsPaid { get; set; }
    public decimal Subtotal { get; set; }
    public decimal DiscountAmount { get; set; }
    public decimal TotalAmount { get; set; }
    public Guid? VoucherId { get; set; }
    public string? VoucherCode { get; set; }
    public List<OrderItem> Items { get; set; } = new();
}
